fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/OrRangePredicate.java

```diff
@@ -0,0 +1,100 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.starrocks.sql.optimizer.rule.transformation.materialization;
+
+import com.starrocks.sql.optimizer.Utils;
+import com.starrocks.sql.optimizer.operator.scalar.ConstantOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+import org.checkerframework.com.google.common.collect.Lists;
+
+import java.util.List;
+import java.util.Objects;
+
+public class OrRangePredicate extends RangePredicate {
+    private List<RangePredicate> rangePredicates;
+
+    public OrRangePredicate(List<RangePredicate> rangePredicates) {
+        this.rangePredicates = rangePredicates;
+    }
+
+    public List<RangePredicate> getRangePredicates() {
+        return rangePredicates;
+    }
+
+    @Override
+    public boolean enclose(RangePredicate other) {
```

Comment:
This `OrRangePredicate ` can enclose `other` predicates?

Reply:
yes, check whether OrRangePredicate can enclose other predicates.